eZ/Bundle/EzPublishCoreBundle/Resources/config/papi.yml

```diff
@@ -75,3 +75,31 @@ services:
             - '@ezpublish.spi.search'
         tags:
             - { name: kernel.event_subscriber }
+
+    eZ\Publish\Core\FieldType\Image\ImageThumbnailStrategy:
+        arguments:
+            $variationHandler: '@ezpublish.image_alias.imagine.variation.imagine_alias_generator'
+            $variationName: 'medium'
+
+    eZ\Publish\Core\FieldType\ImageAsset\ImageAssetThumbnailStrategy:
+        lazy: true
+        arguments:
+            $thumbnailStrategy: '@eZ\Publish\Core\Repository\Strategy\ContentThumbnail\ThumbnailChainStrategy'
+            $contentService: '@ezpublish.api.service.content'
+
+    eZ\Publish\Core\Repository\Strategy\ContentThumbnail\Field\ContentFieldStrategy:
+        arguments:
+            $strategies:
+                ezimage: '@eZ\Publish\Core\FieldType\Image\ImageThumbnailStrategy'
+                ezimageasset: '@eZ\Publish\Core\FieldType\ImageAsset\ImageAssetThumbnailStrategy'
+
+    eZ\Publish\Core\Repository\Strategy\ContentThumbnail\FirstMatchingFieldStrategy:
+        arguments:
+            $fieldTypeService: '@ezpublish.api.service.field_type'
+            $contentFieldStrategy: '@eZ\Publish\Core\Repository\Strategy\ContentThumbnail\Field\ContentFieldStrategy'
+        tags:
+            - { name: kernel.thumbnail_strategy, priority: 0 }
+
+    eZ\Publish\Core\Repository\Strategy\ContentThumbnail\ThumbnailChainStrategy:
+        arguments:
+            $strategies: !tagged_iterator kernel.thumbnail_strategy
```

Comment:
Tags starting with `kernel.` are used by Symfony, so that might be confusing. I still believe `ezplatform.spi.` prefix is a valid one, might be one of:
- `ezplatform.spi.thumbnail_strategy`,
- `ezplatform.spi.kernel.thumbnail_strategy`,
- `ezplatform.spi.content.thumbnail_strategy` (as originally suggested AFAIR).

If for some reason you don't like `spi` here, at least keep `ezplatform` prefix (yes, we already have such tags and don't advocate using `ezpublish` prefix regardless of the surrounding code).